### ISSUE TITLE
Compile 17.1 with SystemUIWithLegacyRecents Makefile

### ISSUE THREAD

**Issue by the-r3dacted** [Flamefire/lineageos_lilac#5]
The entire reason I want to build this is to add this.
https://github.com/kumikooumae/android_vendor_extra

All you need to do is add an .xml with these contents, and it should work
(It's attached to this post inside a .zip because GitHub sucks)
[legacyrecents.zip](https://github.com/Flamefire/lineageos_lilac/files/12599863/legacyrecents.zip)

If you'd rather copy the .xml from the project itself, you need revision="refs/heads/master" to the end for it to sync properly.

**Comment by Flamefire** [comment]
Sorry but I won't do this. I already build & test 6 versions of LOS so I can't add another and bundling extra stuff for everyone which might possibly cause issues or just deviate from base LOS 17.1 could disturb others who want one without this change.

Hence you'll need to build the ROM yourself if you want this on your phone, sorry.

**Comment by the-r3dacted** [comment]
> Sorry but I won't do this. I already build & test 6 versions of LOS so I can't add another and bundling extra stuff for everyone which might possibly cause issues or just deviate from base LOS 17.1 could disturb others who want one without this change.
> 
> Hence you'll need to build the ROM yourself if you want this on your phone, sorry.

I tried that, and it didn't go well.
https://github.com/Flamefire/lineageos_lilac/issues/4

It doesn't need to be an extra version, you could just add it to your normal 17.1 builds. The change doesn't do anything to the stock recents. All it does is add back the old recents as a fallback and is only accessible if the stock Trebuchet launcher is disabled or removed. Nobody would notice it unless they seek it out. Please just add it, I really want to be able to use this ROM.

**Comment by Flamefire** [comment]
> The change doesn't do anything to the stock recents. All it does is add back the old recents as a fallback and is only accessible if the stock Trebuchet launcher is disabled or removed. Nobody would notice it unless they seek it out. 

I'm not so sure about that as this is in the linked readme (emphasis mine)
> This Makefile will **replace** SystemUI module with SystemUIWithLegacyRecents

Maybe I misunderstand this, so is that really an addition and not a replacement?

**Comment by the-r3dacted** [comment]
> > This Makefile will **replace** SystemUI module with SystemUIWithLegacyRecents
> 
> Maybe I misunderstand this, so is that really an addition and not a replacement?

Yeah, the legacy recents were part of SystemUI, so it just adds it back. It will work like Android 9 where you still have the modern recents unless you disable Trebuchet.

**Comment by Flamefire** [comment]
Ok, I'll consider adding it for the January builds (December builds already in testing)

However I won't use that extra repo as it seem to be enough to add `PRODUCT_PACKAGES += SystemUIWithLegacyRecents` in `device/sony/lilac` and the permissions xml just replaces `<permission name="android.permission.CAMERA_OPEN_CLOSE_LISTENER" />` by `<permission name="android.permission.OBSERVE_NETWORK_POLICY"/>` which doesn't look correct.

Can you provide a quick way to test this?

**Comment by the-r3dacted** [comment]
I was going to try compiling, but I realized I don't want to do testing on my main phone.

**Comment by Flamefire** [comment]
I can do the testing on a test phone (for the January builds). Can you provide short and simple steps to check whether the legacy recents view works? I.e. what one should do and what it currently does (or does not) do/show and what it should?

**Comment by the-r3dacted** [comment]
Well just remove Trebuchet (LineageOS launcher) with root, and it should just show up. Maybe you could disable it with QuickSwitch, but I haven't looked at that program that much.

1: Install an alternative launcher.
Some might have a recents menu. I know for a fact Apex Launcher Classic doesn't have one, so probably use that for testing.

2: Remove Trebuchet.
There are many ways to do this, TWRP, using System App Remover, file explorer with root, etc.

3: Reboot

4: Try opening the recents menu.

**Comment by Flamefire** [comment]
A question that come up in the LOS Discord: Why even bother? I.e. why remove Trebuchet if that causes issues with the launcher of your choice? Why instead install another app/app-part, especially as almost no-one will use it?
I'm trying to understand the actual use case here.

**Comment by the-r3dacted** [comment]
I prefer the old recents. You have more apps on screen at one time, and scrolling through them is a lot better and faster.

**Comment by the-r3dacted** [comment]
So apparently I was wrong. I was able to build LineageOS 17.1 for the XZ2C, I added the xml, and it replaces the new recents. I don't see any code for stripping them out of Trebuchet, but I also don't really understand code. My guess is something forces the SystemUI recents to the current. Considering Android 9 had the old recents as a fallback, I think it should be possible to replicate the same functionality with 10, but I'm not sure how.
